# **INTRODUCTION**







This Olympics dataset consists of different major athletic event in which thousands of competitors from all over the world compete in a range of events. Different Nations from all over the world compete and hence contains a lot of data. Using different Data Science and Machine Learning techniques which will be beneficial for different countries different players in their growth.

The following dataset contains information about olympic events that occurred between 2000 and 2012.

The dataset has a total of ten features. There are 8618 observations.

Players' names, ages, countries, medal types, and total medals earned are all included in the data collection.
The dataset also assume Madel won in group sports also as individual Medal for each player.

A single player can play many games.

Year - The year in which a certain olympic event took place.

Date Given - The date on which a sporting event will take place.

Same player can compete in a number of Olympic events.

# **AIM**

Aim of this project is to perform different pyspark techniques on this dataset
such as basic sorting grouping and filtering the data according to different features.

In [ ]:
pip install pyspark

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 281.3 MB 55 kB/s 
     |████████████████████████████████| 199 kB 94.1 MB/s 
  Created wheel for pyspark: filename=pyspark-3.3.0-py2.py3-none-any.whl size=281764026 sha256=7ea536d6f2fe1d98b1c6a04bb921f7d71eb1725e374bc790f678e1c75c073022
  Stored in directory: /root/.cache/pip/wheels/7a/8e/1b/f73a52650d2e5f337708d9f6a1750d451a7349a867f928b885
Successfully built pyspark


In [ ]:

#Initializing PySpark
from pyspark import SparkContext, SparkConf
from pyspark.sql import SparkSession
from pyspark.sql import SQLContext

#Spark Config
conf = SparkConf().setAppName("sample_app")
sc = SparkContext(conf=conf)

sqlContext = SQLContext(sc)

/usr/local/lib/python3.7/dist-packages/pyspark/sql/context.py:114: FutureWarning: Deprecated in 3.0.0. Use SparkSession.builder.getOrCreate() instead.
  FutureWarning,


In [ ]:
from pyspark.sql import SparkSession

spark = (SparkSession
  .builder
  .appName("data")
  .getOrCreate())

# Path to data set
csv_file = "/content/olympix_data_organized_with_header (1) (1) (1).csv"
# Read and create a temporary view
# Infer schema (note that for larger files you 
# may want to specify the schema)
data = (spark.read.format("csv")
  .option("inferSchema", "true")
  .option("header", "true")
  .load(csv_file))
data.createOrReplaceTempView("data")

In [ ]:
# Create the DataFrame 
data = spark.table('data')

# Show the head
data.show()


+--------------------+---+-------------+----+----------+--------------------+----------+------------+-----------+-----------+
|                name|age|      country|year|Date_Given|              sports|gold_medal|silver_medal|brone_medal|total_medal|
+--------------------+---+-------------+----+----------+--------------------+----------+------------+-----------+-----------+
|      Michael Phelps| 23|United States|2008| 8/24/2008|            Swimming|         8|           0|          0|          8|
|      Michael Phelps| 19|United States|2004| 8/29/2004|            Swimming|         6|           0|          2|          8|
|      Michael Phelps| 27|United States|2012|08-12-2012|            Swimming|         4|           2|          0|          6|
|    Natalie Coughlin| 25|United States|2008| 8/24/2008|            Swimming|         1|           2|          3|          6|
|       Aleksey Nemov| 24|       Russia|2000|10-01-2000|          Gymnastics|         2|           1|          3|     

In [ ]:
data.printSchema()

root
 |-- name: string (nullable = true)
 |-- age: integer (nullable = true)
 |-- country: string (nullable = true)
 |-- year: integer (nullable = true)
 |-- Date_Given: string (nullable = true)
 |-- sports: string (nullable = true)
 |-- gold_medal: integer (nullable = true)
 |-- silver_medal: integer (nullable = true)
 |-- brone_medal: integer (nullable = true)
 |-- total_medal: integer (nullable = true)



In [ ]:
from pyspark.sql.functions import col,isnan, when, count
data.select([count(when(isnan(c) | col(c).isNull(), c)).alias(c) for c in data.columns]
   ).show()


+----+---+-------+----+----------+------+----------+------------+-----------+-----------+
|name|age|country|year|Date_Given|sports|gold_medal|silver_medal|brone_medal|total_medal|
+----+---+-------+----+----------+------+----------+------------+-----------+-----------+
|   5|  5|      0|   0|         0|     0|         0|           0|          0|          0|
+----+---+-------+----+----------+------+----------+------------+-----------+-----------+



In [ ]:
data=data.na.drop(subset=["name","age"])

In [ ]:
res1=data.toPandas().info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8613 entries, 0 to 8612
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   name          8613 non-null   object
 1   age           8613 non-null   int32 
 2   country       8613 non-null   object
 3   year          8613 non-null   int32 
 4   Date_Given    8613 non-null   object
 5   sports        8613 non-null   object
 6   gold_medal    8613 non-null   int32 
 7   silver_medal  8613 non-null   int32 
 8   brone_medal   8613 non-null   int32 
 9   total_medal   8613 non-null   int32 
dtypes: int32(6), object(4)
memory usage: 471.1+ KB


In [ ]:

data.exceptAll(data.dropDuplicates(["name","age","country","year","Date_Given","sports","gold_medal","silver_medal","brone_medal","total_medal"]))

DataFrame[name: string, age: int, country: string, year: int, Date_Given: string, sports: string, gold_medal: int, silver_medal: int, brone_medal: int, total_medal: int]

## **The Olympic Sports/games in the dataset.**

In [ ]:

sp=data.select('sports').distinct()

sp.show(49)

+--------------------+
|              sports|
+--------------------+
|          Gymnastics|
|              Tennis|
|              Boxing|
|Short-Track Speed...|
|          Ice Hockey|
|              Rowing|
|                Judo|
|            Softball|
|             Sailing|
|            Swimming|
|       Alpine Skiing|
|          Basketball|
|            Handball|
| Rhythmic Gymnastics|
|            Biathlon|
|           Triathlon|
|           Badminton|
|           Athletics|
|      Figure Skating|
|    Beach Volleyball|
|          Equestrian|
|            Shooting|
|              Diving|
|            Skeleton|
|         Ski Jumping|
|     Nordic Combined|
|           Wrestling|
|       Speed Skating|
|        Table Tennis|
|           Taekwondo|
|            Canoeing|
|       Weightlifting|
|           Waterpolo|
|            Baseball|
|   Modern Pentathlon|
|        Snowboarding|
|           Bobsleigh|
|             Cycling|
|Cross Country Skiing|
|          Volleyball|
|          

## **The total number of medals in each Olympic Sport/game**

In [ ]:
#importing pyspark sql function
from pyspark.sql.functions import sum, col, desc, asc, count
# groupby sports than performing aggregate function on total medals and than arranging in descending order

grp=data.groupBy("sports").agg(sum("total_medal").alias("Total")).sort(desc("Total"))
grp.show()

+----------+-----+
|    sports|Total|
+----------+-----+
|  Swimming|  765|
| Athletics|  753|
|    Rowing|  576|
|  Football|  407|
|    Hockey|  386|
|Ice Hockey|  384|
|  Handball|  351|
|  Canoeing|  333|
| Waterpolo|  306|
|   Cycling|  306|
|Basketball|  287|
|Gymnastics|  282|
|Volleyball|  278|
|   Fencing|  256|
| Wrestling|  245|
|      Judo|  224|
|  Baseball|  216|
|   Sailing|  210|
|  Shooting|  192|
|    Boxing|  188|
+----------+-----+
only showing top 20 rows



## **The total number of medals won by each country in swimming.**

In [ ]:
#groupby sports and country and than aggeregation
new=data.groupby('sports','country').agg(sum("total_medal").alias("Total")).sort(desc("Total"))

In [ ]:
#filter datafram based upon swimming 
new.filter(new['sports']=='Swimming').show()


+--------+-------------+-----+
|  sports|      country|Total|
+--------+-------------+-----+
|Swimming|United States|  267|
|Swimming|    Australia|  163|
|Swimming|  Netherlands|   46|
|Swimming|        Japan|   43|
|Swimming|       France|   39|
|Swimming|        China|   35|
|Swimming|      Germany|   32|
|Swimming|       Russia|   20|
|Swimming|        Italy|   16|
|Swimming| South Africa|   11|
|Swimming|Great Britain|   11|
|Swimming|      Hungary|    9|
|Swimming|       Sweden|    9|
|Swimming|       Brazil|    8|
|Swimming|     Zimbabwe|    7|
|Swimming|      Ukraine|    7|
|Swimming|      Romania|    6|
|Swimming|       Canada|    5|
|Swimming|  South Korea|    4|
|Swimming|        Spain|    3|
+--------+-------------+-----+
only showing top 20 rows



## **The total number of medals won by each country in Skeleton.**

In [ ]:
#filter datafram based upon skeleton
new.filter(new['sports']=='Skeleton').show()

+--------+-------------+-----+
|  sports|      country|Total|
+--------+-------------+-----+
|Skeleton|       Canada|    4|
|Skeleton|United States|    3|
|Skeleton|  Switzerland|    3|
|Skeleton|Great Britain|    3|
|Skeleton|      Germany|    2|
|Skeleton|      Austria|    1|
|Skeleton|       Latvia|    1|
|Skeleton|       Russia|    1|
+--------+-------------+-----+



## **The number of medals that the US won yearly.**

In [ ]:
data.filter(data['country']=='United States').groupby('year','country').agg(sum("total_medal").alias("Total")).sort(desc("year")).show()

+----+-------------+-----+
|year|      country|Total|
+----+-------------+-----+
|2012|United States|  254|
|2010|United States|   97|
|2008|United States|  317|
|2006|United States|   52|
|2004|United States|  265|
|2002|United States|   84|
|2000|United States|  243|
+----+-------------+-----+



## **The total number of medals won by each country.**

In [ ]:
# group by operation on country and aggeregation on total medals
data.groupby('country').agg(sum("total_medal").alias("Total")).sort(desc("Total")).show()

+-------------+-----+
|      country|Total|
+-------------+-----+
|United States| 1312|
|       Russia|  768|
|      Germany|  629|
|    Australia|  609|
|        China|  530|
|       Canada|  370|
|        Italy|  331|
|Great Britain|  322|
|       France|  318|
|  Netherlands|  318|
|  South Korea|  308|
|        Japan|  282|
|       Brazil|  218|
|        Spain|  205|
|       Norway|  192|
|         Cuba|  188|
|       Sweden|  181|
|      Hungary|  145|
|      Ukraine|  143|
|    Argentina|  139|
+-------------+-----+
only showing top 20 rows



## **The oldest athlete in the olympics and the Country which he/she was from**

In [ ]:
# group by country , player name and age , after it max age and sort it by age
df=data.groupby('name','country','age').max('age').sort(desc("max(age)")).limit(1)
df.select('name','country','age').show()

+----------+-------+---+
|      name|country|age|
+----------+-------+---+
|Ian Millar| Canada| 61|
+----------+-------+---+



# **CONCLUSION**



1.  The oldest athlete in the olympics is IAN MILLAR from canada .
2.  The total no of medal won by any country is highest in United States with 1312 total medal and the country Argentina has the lowest no of total medals amongst all i.e 139 medals.
3. The highest no of medals won in swimming category is by United States with 267 medal and the lowest is won by spain with 3 medals.
4. The highest no of medal won is in swimming category and the no is 765 and the lowest no of medal won is i boxing with 188 medal 

